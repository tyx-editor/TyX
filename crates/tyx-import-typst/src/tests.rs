use std::{path::PathBuf, sync::Arc};

use serde::Serialize;
use tinymist_project::LspWorld;

/// Makes snapshot tests for the given function.
/// The fixtures used for snapshot testing are in the `src/fixtures/{name}`
/// directory.
fn snapshot_testing(name: &str, f: &impl Fn(LspWorld, PathBuf)) {
    tinymist_tests::snapshot_testing!(name, |verse, path| {
        f(verse.snapshot(), path);
    });
}

#[test]
fn convert() {
    snapshot_testing("integration", &|world, _path| {
        insta::assert_snapshot!(conv(world));
    });
}

/// Converts the given main document in the world to a string.
fn conv(world: LspWorld) -> String {
    let doc = crate::convert(Arc::new(world));
    // to pretty with indent 1
    let mut w = Vec::new();
    let mut s = serde_json::Serializer::with_formatter(
        &mut w,
        serde_json::ser::PrettyFormatter::with_indent(b" "),
    );
    let node = doc.map(|doc| doc.content);
    node.serialize(&mut s).expect("failed to serialize");
    String::from_utf8(w).expect("invalid utf8")
}
