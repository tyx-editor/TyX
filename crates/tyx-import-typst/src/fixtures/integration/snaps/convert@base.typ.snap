---
source: crates/tyx-import-typst/src/tests.rs
expression: conv(world)
input_file: crates/tyx-import-typst/src/fixtures/integration/base.typ
---
{
 "root": {
  "type": "root",
  "version": 1,
  "children": [
   {
    "type": "heading",
    "version": 1,
    "children": [
     {
      "type": "text",
      "version": 1,
      "detail": 0,
      "format": 0,
      "mode": "normal",
      "style": "",
      "text": "Hello, World!"
     }
    ],
    "direction": null,
    "format": "",
    "indent": 0,
    "tag": "h2"
   },
   {
    "type": "paragraph",
    "version": 1,
    "children": [
     {
      "type": "text",
      "version": 1,
      "detail": 0,
      "format": 0,
      "mode": "normal",
      "style": "",
      "text": "This is a typst document."
     }
    ],
    "direction": null,
    "format": "",
    "indent": 0
   }
  ],
  "direction": null,
  "format": "",
  "indent": 0
 }
}
